### DISCUSSION TITLE
Pre-built binaries for different OSes

### DISCUSSION THREAD

**Discussion by drkameleon** [h3rald/min#97] [Q&A] [answered]
I have noticed that you offer pre-built binaries. And as a matter of fact, I did download the macOS binary and it works beautifully on my Big Sur MacBook Pro.

How do you do it? I've noticed that you have no relevant workflow (@ GitHub Actions), so I guess you're doing it manually? Is it generally reliable?

I'm asking since I've been experimenting a bit with the idea (I mean: serving binaries, instead of asking users to install anything or re-compile everything), but not 100% sure it would work all the time...

Here's a related question I posted earlier today in the Nim forum: 
https://forum.nim-lang.org/t/7398

**Comment by h3rald** [comment, marked as answer]
Well, the answer is [here](https://blog.filippo.io/easy-windows-and-linux-cross-compilers-for-macos/).

Basically, yes, on my Macbook Air I have two cross compilers installed and configured, one for linux and one for windows, and the the of course I compile the macOS one natively.

Yes I *should* run this on a Github workflow but never got around to it. There are macOS runners available so technically it should be possible... although setting up two cross-compiler toolchains can be quite time-consuming.

**Reply by drkameleon** [reply]
Awesome!

My next goal is to do pretty much that, but rather automated (part of the [already existing Release workflow](https://github.com/arturo-lang/arturo/blob/master/.github/workflows/release.yml)) 

Let's see... :)

**Reply by drkameleon** [reply]
Btw, I've just found this - you may want to have a look:

https://github.com/nim-lang/Nim/wiki/BuildServices#8-cross-compilation-and-release

**Reply by h3rald** [reply]
Wonderfuuuuul! Thanks a lot, will definitely check it out!

BTW, in case you run into headaches on linux... if you are looking for a self-contained, statically compiled executable, you may want to comple with musl-gcc, so that musl will be used instead of libc (which otherwise is always dynamically linked)... took me a while to get it to work, but in the end I think it's worth it!

**Reply by drkameleon** [reply]
I know about musl, it's just that I'm a bit cautious with this whole cross-compilation thing and doing it at a rather slow pace.

Btw, I think musl is the default library used by Alpine, so I guess I'm already using it...

I've also create a Docker image for Arturo (based on Alpine):
https://github.com/arturo-lang/art-docker

Generally, super-simple to set up...
